api/src/main/java/marquez/db/JobVersionDao.java

```diff
@@ -453,11 +461,12 @@ private static ExtendedDatasetVersionRow toExtendedDatasetVersionRow(DatasetReco
    * @param transitionedAt The timestamp of the run state transition.
    * @return A {@link BagOfJobVersionInfo} object.
    */
-  default BagOfJobVersionInfo upsertJobVersionOnRunTransition(
+  default BagOfJobVersionInfo upsertJobVersionReferences(
```

Comment:
Do we need to update the method name? `upsertJobVersionReferences()` is far less clear and doesn't provide much in terms of the methods logic.

As an alternative I would introduce `upsertJobVersionForStreaminJob()` that calls then calls `upsertJobVersionOnRunTransition()` if a version should be created.

```java
default upsertJobVersionForStreaminJob(...) {
    final JobVersionDao jobVersionDao = createJobVersionDao();
    if (!jobVersionDao.versionExists(jobVersion.getValue())) {
      upsertJobVersionOnRunTransition(...)
    }
}
```

Then, you can avoid the added logic in `upsertJobVersionOnRunTransition()` to check if a job is a streaming job or not. We can then be more explicit further up the call stack that we are attempting to create a streaming job version by calling `upsertJobVersionForStreaminJob()`.

Note, I'd also provide extensive java docs for `upsertJobVersionForStreaminJob()`.

Reply:
🙌

Reply:
Sure, I refactored this already.